services/galley/src/Galley/API/Teams.hs

```diff
@@ -112,6 +113,8 @@ createNonBindingTeam (zusr::: zcon ::: req ::: _) = do
     let zothers = map (view userId) others
     ensureUnboundUsers (zusr : zothers)
     ensureConnected zusr zothers
+    Log.debug $ Log.field "targets" (toByteString (show zothers))
+              . Log.msg (Log.val "createNonBindingTeam")
```

Comment:
I liked how you scoped `Legalhold...`. How about using `Teams.createNonBindingTeam` here too?

Reply:
I agree but I am too lazy to do this manually, but I have convinced GHC to do it for us instead.  (By stealing some code from the super nice co-log library): here is a small protoype:

```haskell
{-# LANGUAGE ConstraintKinds #-}
{-# LANGUAGE RecordWildCards #-}

import Data.Monoid
import GHC.Stack
import Prelude hiding (log)

data Level = Debug | Info | Warn | Error  deriving Show
showSourceLoc :: CallStack -> String
showSourceLoc cs = showCallStack
  where
    showCallStack :: String
    showCallStack = case getCallStack cs of
        []                             -> "<unknown loc>"
        [(name, loc)]                  -> "lol" -- showLoc name loc
        (_, loc) : (callerName, _) : _ -> showLoc callerName loc
    showLoc :: String -> SrcLoc -> String
    showLoc name SrcLoc{..} =
        srcLocModule <> "." <>  name <> "#" <>  (show srcLocStartLine)
        
class Monad m => MonadLogger m where
    logMsg :: CallStack -> Level -> String -> m ()
    
instance MonadLogger IO where
    logMsg callStack level msg = do
      putStrLn $ "[" ++ show level ++ "] " ++ showSourceLoc callStack ++ " " ++ msg
type MonadLogger' m = (MonadLogger m, HasCallStack)

log :: MonadLogger' m => Level -> String -> m ()
log = withFrozenCallStack (logMsg callStack)

smelt :: MonadLogger' m => m ()
smelt = do
  log Error "NOOO"
  log Warn "Maybe"
  kaas

kaas :: MonadLogger' m => m ()
kaas = do
  log Info "I like cheese"
 
main :: IO ()
main = do
  kaas
  smelt

```

```
[Info] Main.kaas#40 I like cheese
[Error] Main.smelt#34 NOOO
[Warn] Main.smelt#35 Maybe
[Info] Main.kaas#40 I like cheese
```

Reply:
I ended up doing this manually . still think this is a cute thing to have though; ill open an issue for it